nonthermal.cc

```diff
@@ -2683,22 +2739,16 @@ void nt_MPI_Bcast(const int modelgridindex, const int root) {
               MPI_FLOAT, root, MPI_COMM_WORLD);
 
     // communicate NT excitations
-    const auto frac_excitations_list_size_old = nt_solution[modelgridindex].frac_excitations_list.size();
-    auto frac_excitations_list_size_new = nt_solution[modelgridindex].frac_excitations_list.size();
+
+    auto frac_excitations_list_size_new = nt_solution[modelgridindex].frac_excitations_list_size;
     MPI_Bcast(&frac_excitations_list_size_new, 1, MPI_INT, root, MPI_COMM_WORLD);
 
-    if (frac_excitations_list_size_new != frac_excitations_list_size_old) {
-      nt_solution[modelgridindex].frac_excitations_list.resize(frac_excitations_list_size_new);
-    }
+    nt_solution[modelgridindex].frac_excitations_list_size = frac_excitations_list_size_new;
 
-    const auto frac_excitations_list_size = nt_solution[modelgridindex].frac_excitations_list.size();
-    for (size_t excitationindex = 0; excitationindex < frac_excitations_list_size; excitationindex++) {
-      MPI_Bcast(&nt_solution[modelgridindex].frac_excitations_list[excitationindex].frac_deposition, 1, MPI_DOUBLE,
-                root, MPI_COMM_WORLD);
-      MPI_Bcast(&nt_solution[modelgridindex].frac_excitations_list[excitationindex].ratecoeffperdeposition, 1,
-                MPI_DOUBLE, root, MPI_COMM_WORLD);
-      MPI_Bcast(&nt_solution[modelgridindex].frac_excitations_list[excitationindex].lineindex, 1, MPI_INT, root,
-                MPI_COMM_WORLD);
+    if (globals::rank_in_node == 0) {
+      MPI_Bcast(&nt_solution[modelgridindex].frac_excitations_list,
+                nt_solution[modelgridindex].frac_excitations_list_size, MPI_FLOAT, root_node_id,
+                globals::mpi_comm_internode);
```

Comment:
frac_excitations_list is a pointer to a nt_excitation_struct, not a pointer to a float. To do the Bcast, you should be able to set the data type to MPI_BYTE and then set the length to frac_excitations_list_size * sizeof(nt_excitation_struct).